js/src/util/subscribe-to-events.spec.js

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015, 2016 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+import { spy, stub } from 'sinon';
+
+import subscribeToEvents from './subscribe-to-events';
+import {
+  pastLogs, liveLogs, createApi, createContract
+} from './subscribe-to-events.test.js';
+
+const delay = (t) => new Promise((resolve) => {
+  setTimeout(resolve, t);
+});
+
+describe('util/subscribe-to-events', () => {
+  beforeEach(function () {
+    this.api = createApi();
+    this.contract = createContract(this.api);
+  });
+
+  it('installs a filter', async function () {
+    const { api, contract } = this;
+
+    subscribeToEvents(contract, [ 'Foo', 'Bar' ]);
+    await delay(0);
```

Comment:
Not going to grumble, just think that we have a bit more control with actually moving the timer ahead. Well, I've had some success with it. 

https://github.com/ethcore/parity/blob/7cdfaf1a439a5287b55b4195b29878193f0d7049/js/src/3rdparty/shapeshift/shapeshift.spec.js#L173

Either way, what we have works.